src/amlaidatatests/connection.py

```diff
@@ -52,6 +53,20 @@ def connection_factory(default: Optional[str] = None):
     result = urlparse(connection_string)
     kwargs = dict(parse_qsl(result.query))
 
+    # Extract job labels from query string
+    label_keys = [k for k in kwargs.keys() if re.match(r"^labels\.(.*)$", k)]
+    labels = dict(
+        [
+            (re.match(r"^labels\.(.*)$", k)[1], v)
+            for k, v in kwargs.items()
+            if k in label_keys
+        ]
+    )
+    for k in label_keys:
+        kwargs.pop(k)
+    result = result._replace(query=urlencode(kwargs, True))
```

Comment:
Presumably we need to remove the labels from the connection to make ibis work? Does it work without this?

Reply:
Reason for asking is that I *think* it'll ignore this transparently. It means other languages this isn't intended for won't be affected.

The other option is to move the label parsing under the if bigquery logic below.